src/song_loader.cpp

```diff
@@ -0,0 +1,93 @@
+#include "song_loader.h"
+
+SongLoader::SongLoader()
+{
+    std::string dir = std::string(".");
+    check_library = read_song_names(dir, files);
+    if(check_library != false)
+    {
+        shuffle_songs();
+    }  
+}
+
+void SongLoader::notify_listeners()
+{
+    for(I_SongLoaderObserver*listener:listeners) 
+    {
+        listener->current_song(std::string song_path);
+    }
+}
+
+void SongLoader::next_song()
+{
+    current_song_index = (current_song_index + 1)%files.size();
+    files[current_song_index];
+    notify_listeners();
+}
+
+void SongLoader::previous_song()
+{
+    current_song_index = (current_song_index - 1)%files.size();
+    files[current_song_index];
+    notify_listeners();
+}
+
+std::string SongLoader::next_song_name()
+{
+    return files[(current_song_index + 1)%files.size()];
+    notify_listeners();
+}
+
+std::string SongLoader::previous_song_name()
+{
+    return files[(current_song_index - 1)%files.size()]
```

Comment:
The `return` won't do anything.

Maybe you should change the way that `notify_listeners()` works. Maybe it should take a `std::string path`, that way you can specify inside `previous_song_name()` what to "notify_listeners" rather than having `current_song` hardcoded inside the function.